src/leap/mail/mail.py

```diff
@@ -0,0 +1,697 @@
+            raise TypeError
+        part_index = part + 1
+        try:
+            subpart_dict = self._wrapper.get_subpart_dict(
+                part_index)
+        except KeyError:
+            raise TypeError
+        # XXX pass cdocs
+        return MessagePart(subpart_dict)
+
+    # Custom methods.
+
+    def get_tags(self):
+        """
+        """
+        return tuple(self._wrapper.fdoc.tags)
+
+
+class Flagsmode(object):
+    """
+    Modes for setting the flags/tags.
+    """
+    APPEND = 1
+    REMOVE = -1
+    SET = 0
+
+
+class MessageCollection(object):
+    """
+    A generic collection of messages. It can be messages sharing the same
+    mailbox, tag, the result of a given query, or just a bunch of ids for
+    master documents.
+
+    Since LEAP Mail is primarily oriented to store mail in Soledad, the default
+    (and, so far, only) implementation of the store is contained in the
+    Soledad Mail Adaptor, which is passed to every collection on creation by
+    the root Account object. If you need to use a different adaptor, change the
+    adaptor class attribute in your Account object.
+
+    Store is a reference to a particular instance of the message store (soledad
+    instance or proxy, for instance).
+    """
+
+    # TODO LIST
+    # [ ] look at IMessageSet methods
+    # [ ] make constructor with a per-instance deferredLock to use on
+    #     creation/deletion?
+    # [ ] instead of a mailbox, we could pass an arbitrary container with
+    #     pointers to different doc_ids (type: foo)
+    # [ ] To guarantee synchronicity of the documents sent together during a
+    #     sync, we could get hold of a deferredLock that inhibits
+    #     synchronization while we are updating (think more about this!)
+
+    # Account should provide an adaptor instance when creating this collection.
+    adaptor = None
+    store = None
+    messageklass = Message
+
+    def __init__(self, adaptor, store, mbox_indexer=None, mbox_wrapper=None,
+                 count=None):
+        """
+        Constructor for a MessageCollection.
+        """
+        self.adaptor = adaptor
+        self.store = store
+
+        # XXX I have to think about what to do when there is no mbox passed to
+        # the initialization. We could still get the MetaMsg by index, instead
+        # of by doc_id. See get_message_by_content_hash
+        self.mbox_indexer = mbox_indexer
+        self.mbox_wrapper = mbox_wrapper
+
+        # TODO --- review this count shit. I think it's better to patch server
+        # to accept deferreds.
+        # TODO need to initialize count here because imap server does not
+        # expect a defered for the count. caller should return the deferred for
+        # prime_count (ie, initialize) when returning the collection
+        # TODO should increment and decrement when adding/deleting.
+        # TODO recent count should also be static.
+
+        if not count:
+            count = 0
```

Comment:
Why not use `count=0` in the definition of the method instead of doing this if? I don't know if is a pythonism, but I don't like much that things in leap code often have a default value to `None` instead to a default value that has any meaning for the variable.